File: backend/app/schemas/notification.py
from datetime import datetime
from pydantic import BaseModel


class NotificationCreate(BaseModel):
    message: str
    type: str | None = None


class Notification(BaseModel):
    id: int
    message: str
    type: str | None = None
    read: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: efce-ui/src/types/notification.ts
// Copyright (c) 2026 Jeyapragash. All rights reserved.

export type NotificationItem = {
  id: number;
  message: string;
  type?: "info" | "success" | "error";
  read: boolean;
  createdAt: string;
};

export type NotificationCreate = {
  message: string;
  type?: "info" | "success" | "error";
};
